src/__tests__/combinators/sequence.type.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { sequence } from '@combinators'
+import { describe, it, expectTypeOf } from '@testing'
+import { Parser } from '@types'
+
+describe('sequence', () => {
+  type StringParser = Parser<string>
+  type NumberParser = Parser<number>
+
+  it('should have correct inferred signature', () => {
+    expectTypeOf(sequence<[StringParser, StringParser]>).returns.toMatchTypeOf<
+      Parser<[string, string]>
+    >()
+    expectTypeOf(sequence<[StringParser, NumberParser]>).returns.toMatchTypeOf<
+      Parser<[string, number]>
+    >()
+    expectTypeOf(sequence<StringParser[]>).returns.toMatchTypeOf<Parser<string[]>>()
+    expectTypeOf(sequence<NumberParser[]>).returns.toMatchTypeOf<Parser<number[]>>()
+
+    // Fix this ?
+    expectTypeOf(sequence<(NumberParser | StringParser)[]>).returns.toMatchTypeOf<
+      Parser<string[] | number[]>
+    >()
```

Comment:
Should this be runnable with such signature? If not we can try to constraint the usage of unions.

Reply:
I don't see any reason why it wouldn't be runnable, e.g. `sequence` could take an array of `when`s that return either `string` or `number` parsers, or array of `choice`s.